WebDevStudios/wd_s
# WordPress Scripts Upgrade

Closes #731 

### DESCRIPTION

@wordpress/scripts now [supports Webpack 5](https://github.com/WordPress/gutenberg/pull/33818), which means we can finally upgrade some *very* outdated packages _and_ use PostCSS 8!

Here are the Node dependencies that were upgraded:

```bash
 @wordpress/scripts                                            ^17.1.0  →  ^18.0.0     
 copy-webpack-plugin                                             6.4.1  →    9.0.1     
 css-loader                                                     ^5.2.6  →   ^6.2.0     
 css-minimizer-webpack-plugin                                    1.3.0  →    3.0.2     
 eslint-webpack-plugin                                          ^2.5.4  →   ^3.0.1     
 mini-css-extract-plugin                                        ^1.6.2  →   ^2.2.0     
 postcss-loader                                                  4.2.0  →    6.1.1     
 sass-loader                                                    10.2.0  →   12.1.0     
 stylelint-webpack-plugin                                       ^2.2.2  →   ^3.0.1     
 svg-spritemap-webpack-plugin                                    3.9.1  →    4.2.0     
 tailwindcss                                                     2.2.4  →   ^2.2.7
```

**Other**

This was appearing in the terminal and is coming  from the `svg-spritemap` plugin. It's bug [#175](https://github.com/cascornelissen/svg-spritemap-webpack-plugin/issues/175). The plugin author has committed a fix directly to `master`, but has yet to release it under semver.

```bash
"extendDefaultPlugins" utility is deprecated.
Use "preset-default" plugin with overrides instead.
For example:
{
  name: 'preset-default',
  params: {
    overrides: {
      // customize plugin options
      convertShapeToPath: {
        convertArcs: true
      },
      // disable plugins
      convertPathData: false
    }
  }
}
```

I've slapped a band-aid on this by pointing our `package.json` file [directly at the commit hash](https://github.com/WebDevStudios/wd_s/pull/728/commits/0da6f8655f3cff9165a36431647e97f00e5fd8e4) with the fix. This is fugly, but works, and should be considered a temporary until the fix is actually released and properly semver'd.

I've also updated `composer` dependencies and adjusted the `webpack.config` files to use Webpack 5 shape and syntax.

### SCREENSHOTS

Front-end:
![screenshot](https://dl.dropbox.com/s/yh0fhv5dk2rhhl1/Screen%20Shot%202021-08-23%20at%209.31.15%20AM.png?dl=0)

Build:
![screenshot](https://dl.dropbox.com/s/xwd4fdne5lgh64f/Screen%20Shot%202021-08-23%20at%209.31.50%20AM.png?dl=0)

### STEPS TO VERIFY

How do we test this?

1.  `gh pr checkout 728`
2.  `npm ci` 
3.  `npm run build`
4. Verify everything compiles and appears on the front-end correctly
5. `npm run dev`
6. Verify everything in "dev mode" works correctly
7. `npm run lint`
8. Verify linting still works as expected
9. `npm run format`
10. Verify formatting still works as expectedd
